src/rendergraph/opengl/backend/basegeometry.cpp

```diff
@@ -0,0 +1,21 @@
+#include "backend/basegeometry.h"
+
+#include "backend/baseattributeset.h"
+#include "rendergraph/geometry.h"
+
+using namespace rendergraph;
+
+namespace {
+// to mimic sg default
+constexpr auto defaultDrawingMode = DrawingMode::TriangleStrip;
+} // namespace
+
+BaseGeometry::BaseGeometry(
+        const BaseAttributeSet& attributeSet, int vertexCount)
+        : m_pAttributes(attributeSet.attributes().data()),
+          m_attributeCount(attributeSet.attributes().size()),
+          m_sizeOfVertex(attributeSet.sizeOfVertex()),
+          m_drawingMode(defaultDrawingMode),
+          m_vertexCount(vertexCount),
+          m_vertexData(m_vertexCount * m_sizeOfVertex / sizeof(float)) {
```

Comment:
nit: might want to use `vertexCount` instead of `m_vertexCount` in case BaseGeometry's properties were somehow reorder in the feature which could lead to init-ing with a zero value.